include/exec/sequence/iterate.hpp

```diff
@@ -0,0 +1,191 @@
+          __set_value_unless_stopped(static_cast<_Receiver&&>(__self.__op_->__rcvr_));
+        }
+
+        template <same_as<get_env_t> _GetEnv, __decays_to<__t> _Self>
+        friend env_of_t<_Receiver> tag_invoke(_GetEnv, _Self&& __self) noexcept {
+          return stdexec::get_env(__self.__op_->__rcvr_);
+        }
+      };
+    };
+
+    template <class _Range, class _ReceiverId>
+    struct __operation<_Range, _ReceiverId>::__t : __operation_base_t<_Range> {
+      using _Receiver = stdexec::__t<_ReceiverId>;
+      _Receiver __rcvr_;
+
+      using _ItemSender = decltype(stdexec::on(
+        std::declval<trampoline_scheduler&>(),
+        std::declval<__sender_t<_Range>>()));
+
+      using __next_receiver_t = stdexec::__t<__next_receiver<_Range, _ReceiverId>>;
+
+      std::optional<connect_result_t<__next_sender_of_t<_Receiver, _ItemSender>, __next_receiver_t>>
+        __op_{};
+      trampoline_scheduler __scheduler_{};
+
+      void __start_next() noexcept {
+        if (this->__iterator_ == this->__sentinel_) {
+          stdexec::set_value(static_cast<_Receiver&&>(__rcvr_));
+        } else {
+          try {
+            stdexec::start(__op_.emplace(__conv{[&] {
+              return stdexec::connect(
+                exec::set_next(__rcvr_, stdexec::on(__scheduler_, __sender_t<_Range>{this})),
+                __next_receiver_t{this});
+            }}));
+          } catch (...) {
+            stdexec::set_error(static_cast<_Receiver&&>(__rcvr_), std::current_exception());
+          }
+        }
+      }
+
+      template <same_as<__t> _Self>
+      friend void tag_invoke(start_t, _Self& __self) noexcept {
+        __self.__start_next();
+      }
+    };
+
+    template <class _Range>
+    struct __sequence {
+      struct __t {
+        using __id = __sequence;
+        using is_sender = sequence_tag;
+
+        using completion_signatures = stdexec::completion_signatures<
+          set_value_t(std::ranges::range_reference_t<_Range>),
+          set_error_t(std::exception_ptr),
+          set_stopped_t()>;
+
+        STDEXEC_NO_UNIQUE_ADDRESS _Range __range_;
+
+        template <class _Receiver>
+        using __next_receiver_t = stdexec::__t<__next_receiver<_Range, stdexec::__id<_Receiver>>>;
+
+        template < __decays_to<__t> _Self, sequence_receiver_of<completion_signatures> _Receiver>
+          requires sender_to<
+            __next_sender_of_t<_Receiver, __sender_t<_Range>>,
+            __next_receiver_t<_Receiver>
+            >
+        friend auto tag_invoke(subscribe_t, _Self&& __self, _Receiver __rcvr) //
+          noexcept(__nothrow_decay_copyable<_Receiver>)
+            -> stdexec::__t<__operation<_Range, stdexec::__id<_Receiver>>> {
+          return {
+            {std::ranges::begin(__self.__range_), std::ranges::end(__self.__range_)},
+            static_cast<_Receiver&&>(__rcvr)
+          };
+        }
+      };
+    };
+
+    struct iterate_t {
+      template <std::ranges::range _Range>
```

Comment:
From the implementation, I think we need a forward-or-better range.

```suggestion
      template <std::ranges::forward_range _Range>
        requires stdexec::__decay_copyable<_Range>
```